codegen/smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/ServiceWaitersGeneratorTest.kt

```diff
@@ -20,37 +20,37 @@ import software.amazon.smithy.kotlin.codegen.test.*
 import software.amazon.smithy.model.Model
 import software.amazon.smithy.model.shapes.ShapeId
 import kotlin.test.Test
-import kotlin.test.assertFalse
-import kotlin.test.assertTrue
+import kotlin.test.assertEquals
 
 class ServiceWaitersGeneratorTest {
-    private val generated = generateService("simple-service-with-waiter.smithy")
-
     @Test
     fun testServiceGate() {
-        val enabledServiceModel = loadModelFromResource("simple-service-with-waiter.smithy")
-        val enabledServiceSettings = enabledServiceModel.newTestContext().generationCtx.settings
-        assertTrue(ServiceWaitersGenerator().enabledForService(enabledServiceModel, enabledServiceSettings))
-
-        val disabledServiceModel = loadModelFromResource("simple-service.smithy")
-        val disabledServiceSettings = disabledServiceModel.newTestContext().generationCtx.settings
-        assertFalse(ServiceWaitersGenerator().enabledForService(disabledServiceModel, disabledServiceSettings))
+        mapOf(
+            "simple-service-with-operation-waiter.smithy" to true,
+            "simple-service-with-resource-waiter.smithy" to true,
+            "simple-service.smithy" to false,
+        ).forEach { (modelName, expectEnabled) ->
+            val model = loadModelFromResource(modelName)
+            val settings = model.newTestContext().generationCtx.settings
+            val actualEnabled = ServiceWaitersGenerator().enabledForService(model, settings)
+            assertEquals(expectEnabled, actualEnabled)
+        }
     }
 
     @Test
-    fun testMainWaiterMethod() {
+    fun testWaiterSignatureWithOptionalInput() {
         val methodHeader = """
             /**
-             * Wait until a foo exists
+             * Wait until a foo exists with optional input
              */
-            public suspend fun TestClient.waitUntilFooExists(request: DescribeFooRequest = DescribeFooRequest { }): Outcome<DescribeFooResponse> {
+            public suspend fun TestClient.waitUntilFooOptionalExists(request: DescribeFooOptionalRequest = DescribeFooOptionalRequest { }): Outcome<DescribeFooOptionalResponse> {
         """.trimIndent()
         val methodFooter = """
                 val policy = AcceptorRetryPolicy(request, acceptors)
-                return strategy.retry(policy) { describeFoo(request) }
+                return strategy.retry(policy) { describeFooOptional(request) }
             }
         """.trimIndent()
-        generated.shouldContain(methodHeader, methodFooter)
+        generateService().shouldContain(methodHeader, methodFooter)
```

Comment:
small nit: removing the default and being explicit about what model is being generated makes this easier to follow